examples/mir_demo_server/server_example_input_device_config.cpp

```diff
@@ -55,133 +74,227 @@ void me::add_input_device_configuration_options_to(mir::Server& server)
-                                    0.0);
+                                    mir::OptionType::real);
     server.add_configuration_option(touchpad_scroll_speed_scale_opt,
                                     "Scales touchpad scroll events, use negative values for natural scrolling",
-                                    -1.0);
+                                    mir::OptionType::real);
 
     server.add_configuration_option(touchpad_scroll_mode_opt,
-                                    "Select scroll mode for touchpads: [{two-finger, edge}]",
-                                    touchpad_scroll_mode_two_finger);
-
+#ifdef __cpp_lib_format
+                                    std::format("Select scroll mode for touchpads: [{}, {}, {}]",
+                                                touchpad_scroll_mode_edge,
+                                                touchpad_scroll_mode_two_finger,
+                                                touchpad_scroll_mode_button_down_scroll),
+#else
+                                    "Select scroll mode for touchpads: ["s +
+                                    touchpad_scroll_mode_edge + ", "s +
+                                    touchpad_scroll_mode_two_finger + ", "s +
+                                    touchpad_scroll_mode_button_down_scroll + "]"s,
+#endif
+                                    mir::OptionType::string);
     server.add_configuration_option(touchpad_click_mode_opt,
                                     "Select click mode for touchpads: [{area, finger-count}]",
-                                    touchpad_click_mode_finger_count);
+                                    mir::OptionType::integer);
 
-    auto clamp_to_range = [](double val)
+    auto clamp_to_range = [](std::optional<double> opt_val)-> std::optional<double>
     {
-        if (val < -1.0)
-            val = -1.0;
-        else if (val > 1.0)
-            val = 1.0;
-        return val;
+        if (opt_val)
+        {
+            auto val = *opt_val;
+            if (val < -1.0)
+                val = -1.0;
+            else if (val > 1.0)
+                val = 1.0;
+            return std::make_optional(val);
+        }
+        else
+        {
+            return std::nullopt;
+        }
     };
 
-    // TODO the configuration api allows a combination of values. We might want to expose that in the command line api too.
-    auto convert_to_scroll_mode = [](std::string const& val)
+    auto convert_to_scroll_mode = [](std::optional<std::string> const& opt_val)
+    -> std::optional<MirTouchpadScrollMode>
     {
-        if (val == touchpad_scroll_mode_edge)
-            return mir_touchpad_scroll_mode_edge_scroll;
-        if (val == touchpad_scroll_mode_two_finger)
-            return mir_touchpad_scroll_mode_two_finger_scroll;
-        return mir_touchpad_scroll_mode_none;
+        if (opt_val)
+        {
+            if (*opt_val == touchpad_scroll_mode_none)
```

Comment:
This might be easier to read as a `switch`

Reply:
True, but the language only allow integer types in a switch.

Reply:
:facepalm: dart mode. True :+1: